pkg/workflows/diagnostics.go

```diff
@@ -53,6 +54,11 @@ func (s *CollectWorkloadClusterDiagnosticsTask) Name() string {
 
 func (s *CollectMgmtClusterDiagnosticsTask) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
 	logger.Info("collecting management cluster diagnostics")
+	if features.IsActive(features.ExperimentalSelfManagedClusterUpgrade()) {
```

Comment:
I agree the toggle can be on during workload cluster but if the toggle is off, in terms of management cluster, wouldn't the kubeconfig of `commandContext.ManagementCluster` and `commandContext.BootstrapCluster` be different? I guess it would not save the logs of bootstrap cluster if we change `commandContext.BootstrapCluster` to a `commandContext.ManagementCluster`.

Reply:
Management cluster field should point to whichever is the cluster with the management resources at any point in time
Maybe we need to verify that but I think that's how it's supposed to work

Reply:
For the current upgrade workflow, We move management resources to bootstrap cluster after bootstrap cluster creation and it points to a different kubeconfig file. I guess we should keep it to bootstrap cluster for now untill we remove kind upgrade entirely.

Reply:
It seems like we do
https://github.com/aws/eks-anywhere/blob/main/pkg/workflows/upgrade.go#L492